### ISSUE TITLE
[bug] Documented expression for core:required_conan_version in global.conf doesn't work

### ISSUE THREAD

**Issue by jwillikers** [conan-io/conan#9550]
### Environment Details (include every applicable attribute)
  * Operating System+version: Fedora Silverblue 34
  * Compiler+version: GCC 11.2.1-1
  * Conan version: 1.40.0
  * Python version: 3.9.6

### Steps to reproduce (Include if Applicable)

Add the following to `~/.conan/global.conf`.

```
core:required_conan_version = ">=1.30"
```

Now run `conan install` for a project.

```
$ conan install --install-folder=build .
Error in Conan initialization: Current Conan version (1.40.0) does not satisfy the defined one (">=1.30").
```

**Comment by memsharded** [comment]
I think the quotes are to be avoided, have you tried not using them?

**Comment by jwillikers** [comment]
That works! Should the documentation [here](https://docs.conan.io/en/latest/reference/config_files/global_conf.html#global-configuration) remove the double quotes to avoid confusion?

**Comment by memsharded** [comment]
Yes, lets improve that part of the docs, moving this to the docs repo.
